app/scripts/modules/amazon/src/loadBalancer/configure/application/ALBAdvancedSettings.tsx

```diff
@@ -10,36 +10,27 @@ export type IALBAdvancedSettingsProps = IWizardPageProps<IAmazonApplicationLoadB
 class ALBAdvancedSettingsImpl extends React.Component<IALBAdvancedSettingsProps> {
   public static LABEL = 'Advanced Settings';
 
-  public validate() {
-    const errors = {} as FormikErrors<IAmazonApplicationLoadBalancerUpsertCommand>;
-    return errors;
+  public validate(): FormikErrors<IAmazonApplicationLoadBalancerUpsertCommand> {
+    return {};
   }
 
   public render() {
-    const { values } = this.props.formik;
     return (
       <div>
-        <div className="form-group">
-          <div className="col-md-3 sm-label-right">
-            <span>
-              <b>Idle Timeout</b> <HelpField id="loadBalancer.advancedSettings.idleTimeout" />
-            </span>
-          </div>
-          <div className="col-md-4">
-            <Field className="form-control input-sm" type="number" min="0" name="idleTimeout" />
-          </div>
-        </div>
-        <div className="form-group">
-          <div className="col-md-3 sm-label-right">
-            <b>Protection</b> <HelpField id="loadBalancer.advancedSettings.deletionProtection" />
-          </div>
-          <div className="col-md-7 checkbox">
-            <label>
-              <Field type="checkbox" name="deletionProtection" checked={values.deletionProtection} />
-              Enable deletion protection
-            </label>
-          </div>
-        </div>
+        <FormikFormField
+          name="idleTimeout"
+          label="Idle Timeout"
+          help={<HelpField id="loadBalancer.advancedSettings.idleTimeout" />}
+          validate={val => val < 0 && 'Cannot be negative'}
```

Comment:
Agreed, it would be nice.   I considered two options but didn't implement either:

1) `min`/`max` on the `FormField` or `FormikFormField`.  This prop would get passed down to the `NumberInput` and would also automatically add a `Validation.minValue` rule.

  - The `FormikFormField` passes down all extra props down to the `NumberInput` already
  - Typing this is challenging (I would only want to allow valid `NumberInput` props to be set on the `FormikFormField`

```ts
   <FormikFormField<NumberInput>
          name="idleTimeout"
          label="Idle Timeout"
          help={<HelpField id="loadBalancer.advancedSettings.idleTimeout" />}
          max={0} // valid prop because NumberInput generic argument is present
          asdf={0} // invalid prop
          input={NumberInput}
        />
```

The thing I didn't like about this is that I wasn't able to infer the `<FormikFormField<NumberInput>` with Typescript (I'm sure it can be done, I just tried and failed), so we would have to write it explicitly every time.

The other thing I didn't like was that number validation for `min` and `max` would have to be baked into the `FormikFormField` component which shouldn't know anything specifically about number validation IMHO.

The thing that I **did** like about this is you could stick to the shorthand `input={TextInput} placeholder="Your Name"` syntax instead of a prop spread render function like `input={props => <TextInput {...props} placeholder="Your Name />}".  Then again, I *also think* that always using a prop spreading render function everywhere is best for consistency.

2) An Input component can participate in validation.   An Input element could have internal validation rules (like NumberInput knows how to validate `min={0}` and `max={65534}` props, or an `ExpressionInput` knows how to validate expression syntax). 

```ts
   <FormikFormField
          name="idleTimeout"
          label="Idle Timeout"
          help={<HelpField id="loadBalancer.advancedSettings.idleTimeout" />}
          // NumberInput adds a `min` validator to the FormikFormField
          input={props => <NumberInput min={0} {...props} />} 
        />
```

I like this approach better but didn't have time to implement it.  It's a little tricky because of the way these components are composed together.  Essentially the inner `NumberInput` would have to somehow register its validation function with the `FormikFormField`.  It's doable, but tricky to get right I think.

Reply:
Makes perfect sense! Thank you for explaining.